pkg/cmd/pipeline/list_test.go

```diff
@@ -3,38 +3,205 @@ package pipeline
 import (
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/jonboulle/clockwork"
+	"github.com/knative/pkg/apis"
 	"github.com/tektoncd/cli/pkg/testutil"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/reconciler/v1alpha1/pipelinerun/resources"
 	"github.com/tektoncd/pipeline/test"
 	tb "github.com/tektoncd/pipeline/test/builder"
+	corev1 "k8s.io/api/core/v1"
 )
 
-func TestPipelineList(t *testing.T) {
-	names := []string{
-		"tomatoes",
-		"mangoes",
-		"bananas",
+func TestPipelinesList_empty(t *testing.T) {
+
+	cs, _ := test.SeedTestData(test.Data{})
+	p := &testutil.TestParams{Client: cs.Pipeline}
+
+	pipeline := Command(p)
+	output, err := testutil.ExecuteCommand(pipeline, "list", "-n", "foo")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
+	expected := emptyMsg + "\n"
+	if d := cmp.Diff(expected, output); d != "" {
+		t.Errorf("Unexpected output mismatch: %s", d)
+	}
+}
 
-	pipelines := []*v1alpha1.Pipeline{}
-	for _, name := range names {
-		pipelines = append(pipelines, tb.Pipeline(name, "foo", tb.PipelineSpec()))
+func TestPipelineList_only_pipelines(t *testing.T) {
+	pipelines := []pipelineDetails{
+		{"tomatoes", 1 * time.Minute},
+		{"mangoes", 20 * time.Second},
+		{"bananas", 512 * time.Hour}, // 3 weeks
 	}
 
-	cs, _ := test.SeedTestData(test.Data{Pipelines: pipelines})
-	p := &testutil.TestParams{Client: cs.Pipeline}
+	clock := clockwork.NewFakeClock()
+	cs, _ := seedPipelines(clock, pipelines, "namespace")
+	p := &testutil.TestParams{Client: cs.Pipeline, Clock: clock}
 
 	pipeline := Command(p)
-	output, err := testutil.ExecuteCommand(pipeline, "list", "-n", "foo")
+	output, err := testutil.ExecuteCommand(pipeline, "list", "-n", "namespace")
 	if err != nil {
 		t.Errorf("Unexpected error: %v", err)
 	}
 
-	//  account for new line at the end
-	expected := strings.Join(append(names, ""), "\n")
-	if d := cmp.Diff(expected, output); d != "" {
+	expected := []string{
+		"NAME       AGE          LAST RUN   STARTED   DURATION   STATUS",
+		"tomatoes   1 minute     ---        ---       ---        ---",
```

Comment:
@siamaksade , do you feel it would look better with `ago`? 

e.g.

```
$ tkn pipeline ls -n test-tekton

NAME              AGE         LAST RUN                STARTED     DURATION     STATUS
output-pipeline   2 days ago   output-pipeline-run-5   1 day ago   27 seconds   Succeeded
```